### PR TITLE
Memory Target

### LINKED ISSUE #54: Memory Target
To make `Copy` useful, we need to implement a `Memory` target based on the current `oras.land/oras-go/pkg/content.Memory`.

### PR DESCRIPTION
**Pull request by shizhMSFT**
- Resolves #54.
- Add test for `oras.Copy()` using the `Memory` target.

### CODE REVIEW

File: content/memory/memory.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+	upEdgeKey := descriptor.FromOCI(expected)
+	downEdges, err := content.DownEdges(ctx, s.storage, expected)
+	if err != nil {
+		return err
+	}
+	for _, downEdge := range downEdges {
+		downEdgeKey := descriptor.FromOCI(downEdge)
+		upEdgesValue, _ := s.upEdges.LoadOrStore(downEdgeKey, &sync.Map{})
+		upEdges := upEdgesValue.(*sync.Map)
+		upEdges.Store(upEdgeKey, expected)
```
Comment (Wwwsylvia) [inline content/memory/memory.go:69]: The `upEdgeValue` variable naming is confusing. It looks like a value of the upEdges and is paired with `upEdgeKey`, but it is not...

Comment (shizhMSFT) [reply content/memory/memory.go]: Let's reduce the confusion by just naming it `value`.

**Review by Wwwsylvia — APPROVED** [review]
LGTM

**Comment by sajayantony** [comment]
I believe this addreses @deitch's earlier questions. Would be good to have an LGTM from @deitch

**Review by deitch — APPROVED** [review]
LGTM. I compared it to what we have, I think I rather like how this is structured. I spent some time on the internal structure, but mostly seeing functionality via the good tests.

**Comment by deitch** [comment]
I was going to merge this in, but don't know if you are ready for it @shizhMSFT .

**Comment by shizhMSFT** [comment]
@deitch I'm ready to merge it.

**Pull request merged**